cadence/contracts/bridge/FlowEVMBridge.cdc

```diff
@@ -168,6 +188,144 @@ contract FlowEVMBridge : IFlowEVMNFTBridge, IFlowEVMTokenBridge {
         self.deployDefiningContract(evmContractAddress: address)
     }
 
+    /// Registers a custom cross-VM NFT implementation, allowing projects to integrate their Cadence & EVM contracts
+    /// such that the VM bridge facilitates movement between VMs as the integrated implementations.
+    ///
+    /// @param type: The NFT Type to register as cross-VM NFT
+    /// @param fulfillmentMinter: The optional NFTFulfillmentMinter Capability. This parameter is required for
+    ///     EVM-native NFTs
+    /// @param feeProvider: A reference to a FungibleToken Provider from which the bridging fee is withdrawn in $FLOW
+    ///
+    access(all)
+    fun registerCrossVMNFT(
```

Comment:
In writing another comment, I'm realizing we'll need an additional conditional in registration. A spoofer could permissionlessly register a cross-VM NFT with a `NFTFulfillmentMinter` Capability that is not associated with the project, effectively front-running the cross-VM project's registration and preventing them from properly registering. That condition will need to be `type.address == fulfillmentMinterCap!.borrow().getType().address` to ensure the minter is not defined in a third-party account. Otherwise we run the aforementioned risk.

Reply:
Fixed in [6a6bc3a](https://github.com/onflow/flow-evm-bridge/pull/173/commits/6a6bc3a05ae5e0375a5ad7c12c6a6392ad251faf) with test case